Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -209,8 +210,60 @@ CFArrayRef _DWriteCopyFontNamesForFamilyName(CFStringRef familyName) {
  * Note: This function currently uses a cache, meaning that fonts installed during runtime will not be reflected,
  * unless registered to the user font collection
  */
-CFStringRef _DWriteGetFamilyNameForFontName(CFStringRef fontName) {
-    return _DWriteGetFontPropertiesFromName(fontName)->familyName.get();
+CFStringRef _DWriteGetCompatibleFamilyName(CFStringRef fontName, IDWriteFontFace* fontFace) {
+    woc::unique_cf<CFStringRef> upperFontName(_CFStringCreateUppercaseCopy(fontName));
+
+    const auto& info = __GetSystemFontCollectionHelper()->GetFontPropertiesFromUppercaseFontName(upperFontName);
+    if (info) {
+        return info->familyName.get();
+    }
+
+    const auto& userFontInfo = __GetUserFontCollectionHelper()->GetFontPropertiesFromUppercaseFontName(upperFontName);
+    if (userFontInfo) {
+        if (!fontFace) {
+            return userFontInfo->familyName.get();
+        }
+
+        // Retrieve the font file that was used to create this font face
+        UINT32 fileCount;
+        fontFace->GetFiles(&fileCount, nullptr);
+        RETURN_RESULT_IF(fileCount == 0, userFontInfo->familyName.get());
+        if (fileCount > 1) {
+            CFIndex familyNameLength = CFStringGetLength(userFontInfo->familyName.get());
+            std::vector<UniChar> chars(familyNameLength + 1);
+            CFStringGetCharacters(userFontInfo->familyName.get(), { 0, familyNameLength }, chars.data());
+            TraceWarning(TAG, L"The font %ls has %d files, defaulting to the first", chars.data(), fileCount);
+            fileCount = 1;
+        }
+
+        ComPtr<IDWriteFontFile> fontFile;
+        RETURN_RESULT_IF_FAILED(fontFace->GetFiles(&fileCount, &fontFile), userFontInfo->familyName.get());
+
+        // Get the CGDataProvider used to register this font
+        // Note that GetReferenceKey will set fontProvider to a pointer to the file's internal CGDataProviderRef
+        CGDataProviderRef* fontProvider = nullptr;
+        UINT32 keySize;
+        RETURN_RESULT_IF_FAILED(fontFile->GetReferenceKey((const void**)&fontProvider, &keySize), userFontInfo->familyName.get());
+        RETURN_RESULT_IF(keySize != sizeof(fontProvider) || !fontProvider || !(*fontProvider), userFontInfo->familyName.get());
```

Comment:
if so be consistent 

---
In reply to: [107769673](https://github.com/Microsoft/WinObjC/pull/2297#discussion_r107769673) [](ancestors = 107769673)